### ISSUE TITLE
Blockchain silently being forked

### ISSUE THREAD

**Issue by gjfrn** [iron-fish/ironfish#731]
Since running 0.1.10, my node appears to have forked the blockchain (log below). More guidance should be given to the user regarding resolution of this issue, as the node simply carries on running. 

yarn start status -f makes no indication as to this issue.

Probably as a consequence of this, I'm seeing a lot of errors such as:
Invalid block adding to fork 00000...1f606 (18570): Transaction expired

Also, mined blocks present with the following error:

Successfully mined block 0000000000a1a616bfcf60e302f0aea137af11e4cdf227bc654c4a444c0954a8 (18975) has 11 transactions
Failed to add mined block to chain with reason error

Successfully mined block 0000000023ed6208306135c86418b31b8fc1a4edbadb6fc62563616702c86bc4 (18975) has 11 transactions
Failed to add mined block to chain with reason error

Successfully mined block 000000000d4d4624ee715558fb832f855267b130f1a8970f43855738479d10af (18975) has 11 transactions
Failed to add mined block to chain with reason error

Successfully mined block 00000000160e79ee2620008f6d3e6dfec59cb98d8f99cc324c887313d709de5c (18975) has 11 transactions
Failed to add mined block to chain with reason error

Successfully mined block 000000001f0876220d9e513dee7211670c31db57363e8f306d541c6ea7245e48 (18975) has 11 transactions
Failed to add mined block to chain with reason error

Successfully mined block 00000000220e95360029a1cfb971d9d2c49772c69f51c7f0d4acd415f9c0cac5 (19002) has 11 transactions
Failed to add mined block to chain with reason error

Successfully mined block 00000000246e5c08833d015842369e24ad3ca0a0fbc20b395b3e2e15a38d619b (19002) has 11 transactions
Failed to add mined block to chain with reason error

Successfully mined block 000000003e5c18d0f76119bcbcb396b84b17f5b07fca26459761a30b17b35d9d (19062) has 11 transactions
Failed to add mined block to chain with reason error

Successfully mined block 00000000090ee20ac8426266b8f73819a9f3ee6c7483a93f6f7c951b31e2f69b (19082) has 11 transactions
Failed to add mined block to chain with reason error

Successfully mined block 00000000202dd5474560b13069489e326439fb78e03bcf46a336892ba0799009 (19124) has 11 transactions
Failed to add mined block to chain with reason error

---

::::::::::          :::::::::::::::::         
::::::::::::       :::::::::::::::::::        
:::::::::::::     :::::::::::::::::::::       
::::::::::::::   ::::::::::::::::::::::::     
 ::::::::::::: ::::::::::        :::::::::    
   :::::::::::::::::::::          ::::::::::  
   :::::::::::::::::::::          :::::::::   
 ::::::::::::: ::::::::::        :::::::::    
::::::::::::::   ::::::::::::::::::::::::     
:::::::::::::     :::::::::::::::::::::       
::::::::::::       :::::::::::::::::::        
::::::::::           ::::::::::::::::         

Version       0.1.10 @ src
Node Name     NONE
Peer Identity FkquuRvfbLMBEcQV+htqBRcRRQiP2LXjy6hNByoQOzs=
Peer Agent    if/cli/src
Peer Port     9033
Bootstrap     test.bn1.ironfish.network
 
WebSocket server started at :::9033
Connected to the Iron Fish network
Peer URh+tc2 sent orphan 00000...549ec (19254), syncing orphan chain.
Starting sync from URh+tc2. work: +4241608962309, ours: 18974, theirs: 19253
Finding ancestor using linear search on last 3 blocks starting at 00000...33cbc (18974) from peer URh+tc2 at 19253
Found peer URh+tc2 ancestor 00000...33cbc, syncing from 18974 -> 19254 (280) after 1 requests
Requesting 20 blocks starting at 00000...33cbc (18974) from URh+tc2
Peer URh+tc2 disconnected (DISCONNECTED) while syncing.
Added block to fork seq: 16510, head-seq: 18974, hash: 00000...46f08, head-hash: 00000...33cbc, work: 1252423423083877, head-work: 1319605901019302, work-diff: 67182477935425
Peer I/KAZbp sent orphan 00000...6055a (19137), syncing orphan chain.
Starting sync from I/KAZbp. work: +4253686168680, ours: 18974, theirs: 19254
Finding ancestor using linear search on last 3 blocks starting at 00000...33cbc (18974) from peer I/KAZbp at 19254
Found peer I/KAZbp ancestor 00000...33cbc, syncing from 18974 -> 19254 (280) after 1 requests
Requesting 20 blocks starting at 00000...33cbc (18974) from I/KAZbp
Peer I/KAZbp disconnected (DISCONNECTED) while syncing.
Peer 6bmRwDH (orendamir) sent orphan 00000...89a82 (19255), syncing orphan chain.
Starting sync from 6bmRwDH (orendamir). work: +4253686168680, ours: 18974, theirs: 19254
Finding ancestor using linear search on last 3 blocks starting at 00000...33cbc (18974) from peer 6bmRwDH (orendamir) at 19254
Found peer 6bmRwDH (orendamir) ancestor 00000...33cbc, syncing from 18974 -> 19255 (281) after 1 requests
Requesting 20 blocks starting at 00000...33cbc (18974) from 6bmRwDH (orendamir)
Peer 6bmRwDH (orendamir) disconnected (DISCONNECTED) while syncing.
Peer +R7/8Zk sent orphan 00000...4b1c5 (19255), syncing orphan chain.
Starting sync from +R7/8Zk. work: +4241608962309, ours: 18974, theirs: 19253
Finding ancestor using linear search on last 3 blocks starting at 00000...33cbc (18974) from peer +R7/8Zk at 19253
Found peer +R7/8Zk ancestor 00000...33cbc, syncing from 18974 -> 19255 (281) after 1 requests
Requesting 20 blocks starting at 00000...33cbc (18974) from +R7/8Zk
Peer +R7/8Zk disconnected (DISCONNECTED) while syncing.
Added block to fork seq: 18573, head-seq: 18974, hash: 00000...8068b, head-hash: 00000...33cbc, work: 1308141820239832, head-work: 1319605901019302, work-diff: 11464080779470
Peer +/qGhuM sent orphan 00000...7d138 (19256), syncing orphan chain.
Starting sync from +/qGhuM. work: +4253686168680, ours: 18974, theirs: 19254
Finding ancestor using linear search on last 3 blocks starting at 00000...33cbc (18974) from peer +/qGhuM at 19254
Found peer +/qGhuM ancestor 00000...33cbc, syncing from 18974 -> 19256 (282) after 1 requests
Requesting 20 blocks starting at 00000...33cbc (18974) from +/qGhuM
Peer +/qGhuM disconnected (DISCONNECTED) while syncing.
Invalid block adding to fork 00000...1f606 (18570): Transaction expired
Peer As3V6EG sent an invalid block. score: 10, hash: 00000...1f606 (
          18570
        ), reason: Transaction expired
Peer As3V6EG has been banned: Transaction expired
Peer b6hzwAF (slavasprava) sent orphan 00000...4f42d (19257), syncing orphan chain.
Starting sync from b6hzwAF (slavasprava). work: +4253686168680, ours: 18974, theirs: 19254
Finding ancestor using linear search on last 3 blocks starting at 00000...33cbc (18974) from peer b6hzwAF (slavasprava) at 19254
Found peer b6hzwAF (slavasprava) ancestor 00000...33cbc, syncing from 18974 -> 19257 (283) after 1 requests
Requesting 20 blocks starting at 00000...33cbc (18974) from b6hzwAF (slavasprava)
Peer b6hzwAF (slavasprava) disconnected (DISCONNECTED) while syncing.
Peer TsPEErp (mangemonsushi) sent orphan 00000...d4ec3 (19258), syncing orphan chain.
Starting sync from TsPEErp (mangemonsushi). work: +4241608962309, ours: 18974, theirs: 19253
Finding ancestor using linear search on last 3 blocks starting at 00000...33cbc (18974) from peer TsPEErp (mangemonsushi) at 19253
Found peer TsPEErp (mangemonsushi) ancestor 00000...33cbc, syncing from 18974 -> 19258 (284) after 1 requests
Requesting 20 blocks starting at 00000...33cbc (18974) from TsPEErp (mangemonsushi)
Peer TsPEErp (mangemonsushi) disconnected (DISCONNECTED) while syncing.
Peer 16ZQvZp (thomassoo) sent orphan 00000...71f0c (19259), syncing orphan chain.
Starting sync from 16ZQvZp (thomassoo). work: +4253686168680, ours: 18974, theirs: 19254
Finding ancestor using linear search on last 3 blocks starting at 00000...33cbc (18974) from peer 16ZQvZp (thomassoo) at 19254
Found peer 16ZQvZp (thomassoo) ancestor 00000...33cbc, syncing from 18974 -> 19259 (285) after 1 requests
Requesting 20 blocks starting at 00000...33cbc (18974) from 16ZQvZp (thomassoo)
Peer 16ZQvZp (thomassoo) disconnected (DISCONNECTED) while syncing.
Peer BcbpoS9 (iddaddettam) sent orphan 00000...e6d7a (19260), syncing orphan chain.
Starting sync from BcbpoS9 (iddaddettam). work: +4253686168680, ours: 18974, theirs: 19254
Finding ancestor using linear search on last 3 blocks starting at 00000...33cbc (18974) from peer BcbpoS9 (iddaddettam) at 19254
Found peer BcbpoS9 (iddaddettam) ancestor 00000...33cbc, syncing from 18974 -> 19260 (286) after 1 requests
Requesting 20 blocks starting at 00000...33cbc (18974) from BcbpoS9 (iddaddettam)
Peer BcbpoS9 (iddaddettam) disconnected (DISCONNECTED) while syncing.
Peer GzMGFn5 (sobolkzn) sent orphan 00000...9a43e (19261), syncing orphan chain.
Starting sync from GzMGFn5 (sobolkzn). work: +4277763962664, ours: 18974, theirs: 19256
Finding ancestor using linear search on last 3 blocks starting at 00000...33cbc (18974) from peer GzMGFn5 (sobolkzn) at 19256
Found peer GzMGFn5 (sobolkzn) ancestor 00000...33cbc, syncing from 18974 -> 19261 (287) after 1 requests
Requesting 20 blocks starting at 00000...33cbc (18974) from GzMGFn5 (sobolkzn)
Peer GzMGFn5 (sobolkzn) disconnected (DISCONNECTED) while syncing.
Invalid block adding to fork 00000...fd7f0 (18112): Transaction expired

**Comment by CumpsD** [comment]
So, what is the resolution? :)

**Comment by ovsienko** [comment]
I have similar issue. 
Version       0.1.10 @ 191d772
Node Name     Sitch
Peer Identity +E4AsFwQ6rc3os6iAtOlVw2/sPZPXoQbggpDcyqw9l4=
Peer Agent    if/cli/191d772
Peer Port     9033
Bootstrap     test.bn1.ironfish.network
 
WebSocket server started at :::9033
Connected to the Iron Fish network
Starting sync from 6VJGXW9 (www_node1). work: +64602756215173, ours: 16684, theirs: 19078
Finding ancestor using linear search on last 3 blocks starting at 00000...a5483 (16684) from peer 6VJGXW9 (www_node1) at 19078
Found peer 6VJGXW9 (www_node1) ancestor 00000...a5483, syncing from 16684 -> 19078 (2394) after 1 requests
Requesting 20 blocks starting at 00000...a5483 (16684) from 6VJGXW9 (www_node1)
Peer 6VJGXW9 (www_node1) disconnected (DISCONNECTED) while syncing.
Starting sync from 4LDfQAQ (megapapka). work: +61858302588764, ours: 16684, theirs: 18925
Finding ancestor using linear search on last 3 blocks starting at 00000...a5483 (16684) from peer 4LDfQAQ (megapapka) at 18925
Found peer 4LDfQAQ (megapapka) ancestor 00000...a5483, syncing from 16684 -> 18925 (2241) after 1 requests
Requesting 20 blocks starting at 00000...a5483 (16684) from 4LDfQAQ (megapapka)
Peer 4LDfQAQ (megapapka) disconnected (DISCONNECTED) while syncing.
Starting sync from SPlNVz0 (Pick_Nick). work: +49231270844643, ours: 16684, theirs: 18525
Finding ancestor using linear search on last 3 blocks starting at 00000...a5483 (16684) from peer SPlNVz0 (Pick_Nick) at 18525
Found peer SPlNVz0 (Pick_Nick) ancestor 00000...a5483, syncing from 16684 -> 18525 (1841) after 1 requests
Requesting 20 blocks starting at 00000...a5483 (16684) from SPlNVz0 (Pick_Nick)
Peer SPlNVz0 (Pick_Nick) disconnected (DISCONNECTED) while syncing.
Starting sync from sLGBFNg (FamiliarFace). work: +56987338379007, ours: 16684, theirs: 18737
Finding ancestor using linear search on last 3 blocks starting at 00000...a5483 (16684) from peer sLGBFNg (FamiliarFace) at 18737
Found peer sLGBFNg (FamiliarFace) ancestor 00000...a5483, syncing from 16684 -> 18737 (2053) after 1 requests
Requesting 20 blocks starting at 00000...a5483 (16684) from sLGBFNg (FamiliarFace)
Peer sLGBFNg (FamiliarFace) disconnected (DISCONNECTED) while syncing.
Starting sync from IFK9mwV (MemberIT). work: +72804781481499, ours: 16684, theirs: 19717
Finding ancestor using linear search on last 3 blocks starting at 00000...a5483 (16684) from peer IFK9mwV (MemberIT) at 19717
Found peer IFK9mwV (MemberIT) ancestor 00000...a5483, syncing from 16684 -> 19717 (3033) after 1 requests
Requesting 20 blocks starting at 00000...a5483 (16684) from IFK9mwV (MemberIT)
Peer IFK9mwV (MemberIT) disconnected (DISCONNECTED) while syncing.
Starting sync from k/6y+O5 (Shizik101). work: +64757195969968, ours: 16684, theirs: 19088
Finding ancestor using linear search on last 3 blocks starting at 00000...a5483 (16684) from peer k/6y+O5 (Shizik101) at 19088
Found peer k/6y+O5 (Shizik101) ancestor 00000...a5483, syncing from 16684 -> 19088 (2404) after 1 requests
Requesting 20 blocks starting at 00000...a5483 (16684) from k/6y+O5 (Shizik101)

I deleted folder "databases" but when sync has reached the block 16684 i got an error again.
